app/controllers/admin/categories_controller.rb

```diff
@@ -25,12 +25,17 @@ def destroy
 
   def new_or_edit
     @categories = Category.find(:all)
-    @category = Category.find(params[:id])
+    @category = case params[:id]
```

Comment:
Sounds good. I originally chose the case statement because I wanted to stay consistent with how other new_or_edit methods are defined in the project.